homeassistant/components/august/gateway.py

```diff
@@ -34,14 +34,15 @@
 class AugustGateway:
     """Handle the connection to August."""
 
+    api: ApiAsync
```

Comment:
Most methods assume that `api` is set (which happens in `async_setup`). Moved it here to not have to deal with `optional`.

https://github.com/home-assistant/core/blob/5f08e2a2d17d0d6f308d5f22ddff51e427b4b42e/homeassistant/components/august/gateway.py#L73-L87

Feel free to merge if that's fine.

--
Will likely need to do something similar for other variables as well.

Reply:
I figured that was the thought process here.  Its fine with me